core/touch_gesture.js

```diff
@@ -257,32 +265,44 @@ Blockly.TouchGesture.prototype.handleTouchMove = function(e) {
   this.cachedPoints_[pointerId] = this.getTouchPoint(e);
 
   var pointers = Object.keys(this.cachedPoints_);
-  // If two pointers are down, check for pinch gestures
-  if (pointers.length == 2) {
-    // Calculate the distance between the two pointers
-    var point0 = /** @type {!Blockly.utils.Coordinate} */ (
-      this.cachedPoints_[pointers[0]]);
-    var point1 = /** @type {!Blockly.utils.Coordinate} */ (
-      this.cachedPoints_[pointers[1]]);
-    var moveDistance = Blockly.utils.Coordinate.distance(point0, point1);
-    var startDistance = this.startDistance_;
-    var scale = this.touchScale_ = moveDistance / startDistance;
+  if (this.isPinchZoomEnabled_ && pointers.length === 2) {
+    this.handlePinch(e);
+  } else {
+    Blockly.TouchGesture.superClass_.handleMove.call(this, e);
+  }
+};
+
+/**
+* Handle pinch zoom gesture.
+* @param {!Event} e A touch move, or pointer move event.
+* @package
+*/
+Blockly.TouchGesture.prototype.handlePinch = function(e) {
+  var pointers = Object.keys(this.cachedPoints_);
+  // Calculate the distance between the two pointers
+  var point0 = /** @type {!Blockly.utils.Coordinate} */ (
+    this.cachedPoints_[pointers[0]]);
+  var point1 = /** @type {!Blockly.utils.Coordinate} */ (
+    this.cachedPoints_[pointers[1]]);
+  var moveDistance = Blockly.utils.Coordinate.distance(point0, point1);
+  var startDistance = this.startDistance_;
+  this.touchScale_ = moveDistance / startDistance;
 
-    if (this.previousScale_ > 0 && this.previousScale_ < Infinity) {
-      var gestureScale = scale - this.previousScale_;
-      var delta = gestureScale > 0 ?
+  if (this.previousScale_ > 0 && this.previousScale_ < Infinity) {
+    var gestureScale = this.touchScale_ - this.previousScale_;
```

Comment:
Curious why scale is now stored on the object? I don’t see it read anywhere outside this method

Reply:
It was always stored on the object (see old L269). Given its a private field and I also don't see it anywhere outside of this method, I'm going to change it to a local variable and if we need it in future, we can update it then.